### ISSUE TITLE
Support Ratpack 1.6

### ISSUE THREAD

**Issue by llinder** [hyleung/ratpack-zipkin#64]
Ratpack 1.6 is out and should include a client interceptor API now.   I suspect we will get better milage from using the interceptors and hope that it may also fix some anomalies around client tracing that have been seen under heavy load.

**Comment by hyleung** [comment]
@llinder cool! From #62 (man, that was a long time ago), there were a few tweaks required to even get up to 1.5.4 but if we can jump straight to 1.6 then let's do that as a first step before #44 - when I first started looking at that issue (when I was using the 1.6 snapshot) I hit up against some issues around stuff executing on non-ratpack-managed threads. Tho'...that was a long time ago so 🤞

**Comment by llinder** [comment]
Thanks for the heads up on the issues.  So far the upgrade has been going ok.  I actually started on the client interceptor changes first.  My biggest concern right now is backwards compatibility with Ratpack 1.5.x.  I guess at worst case we end up with 2 branches, either that or we try to be clever on detecting the version and maintain different code internally.  Either way its not going to be the most pleasant I suppose.

Sorry I skipped the 1.5.4 upgrade you started.  We have been seeing weird things with client tracing where 2 spans are started on occasion and only one is finished.  My hope is that the 1.6 interceptors resolve this... if not then we can raise an issue in Ratpack.
